cypress/support/auth-provider-commands/cognito.ts

```diff
@@ -46,3 +46,55 @@ Cypress.Commands.add("loginByCognitoApi", (username, password) => {
 
   cy.visit("/");
 });
+
+// Amazon Cognito
+Cypress.Commands.add("loginByCognito", (username, password) => {
+  cy.session(
+    `cognito-${username}`,
+    () => {
+      Cypress.log({
+        displayName: "COGNITO LOGIN",
+        message: [`🔐 Authenticating | ${username}`],
+        // @ts-ignore
+        autoEnd: false,
+      });
+
+      cy.visit("/");
+      cy.contains("Sign in with AWS", {
+        includeShadowDom: true,
+      }).click();
+
+      cy.origin(
+        Cypress.env("cognito_domain"),
+        {
+          args: {
+            username,
+            password,
+          },
+        },
+        ({ username, password }) => {
+          // cognito log in page has some elements of the same id but are off screen. we only want the visible elements to log in
+          cy.get('input[name="username"]:visible').type(username);
+          cy.get('input[name="password"]:visible').type(password, {
+            log: false,
+          });
+          cy.get('input[name="signInSubmitButton"]:visible').click();
+        }
+      );
+
+      // give a few seconds for redirect to settle
+      // eslint-disable-next-line cypress/no-unnecessary-waiting
+      cy.wait(2000);
+
+      // verify we have made it passed the login screen
+      cy.contains("Get Started").should("be.visible");
+    },
+    {
+      validate() {
+        cy.visit("/");
+        // revalidate our session to make sure we are logged in
+        cy.contains("Get Started").should("be.visible");
```

Comment:
Is there another option for validating this was setup correctly?

Reply:
Could you make a request to a protected route and insure it returns a 200?